pkg/cluster/meta/meta.go

```diff
@@ -13,8 +13,59 @@
 
 package meta
 
-import "github.com/joomcode/errorx"
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/joomcode/errorx"
+)
 
 var (
 	errNS = errorx.NewNamespace("meta")
 )
+
+// error types
+const (
+	errTypeConflict = "conflict"
+	errTypeMismatch = "mismatch"
+)
+
+// ValidateErr is the error when meta validation fails with conflicts
+type ValidateErr struct {
+	ty     string      // conflict type
+	target string      // conflict target
+	value  interface{} // conflict value
+	one    string      // object 1
+	two    string      // object 2
+}
+
+// Error implements the error interface
+func (e *ValidateErr) Error() string {
+	return fmt.Sprintf("%s %s for '%v' between '%s' and '%s'", e.target, e.ty, e.value, e.one, e.two)
+}
+
+// Unwrap implements the error interface
+func (e *ValidateErr) Unwrap() error { return nil }
+
+// Is implements the error interface
+func (e *ValidateErr) Is(target error) bool {
```

Comment:
Can you add some tests for this interface implementation?

Reply:
Does type assertion not enough?

Reply:
No, if two error objects has different values, they should be considered different errors